app/Models/Beatmapset.php

```diff
@@ -411,7 +412,10 @@ public static function search(array $params = [])
 
         if (count($beatmap_ids) > 0) {
             $ids = implode(',', $beatmap_ids);
-            $beatmaps = self::whereIn('beatmapset_id', $beatmap_ids)->orderByRaw(DB::raw("FIELD(beatmapset_id, {$ids})"))->get();
+            $beatmaps = self::whereIn('beatmapset_id', $beatmap_ids)
+                ->with('favorites')
```

Comment:
this doesn't help the `hasFavorite` method (because the method doesn't use `$this->favorites` but a different query). And probably loads a bit too many data this way anyway.